File: examples/python-security-fail/src/order_query_service.py
import logging
import subprocess


class OrderQueryService:
    def execute(self, user_id: str, password: str, token: str) -> dict[str, str]:
        table_name = "orders"
        sql = "SELECT * FROM orders WHERE user_id = '" + user_id + "'"
        delete_sql = f"DELETE FROM orders WHERE user_id = '{user_id}'"
        api_key = "hard-coded-secret"
        logging.info("password=%s token=%s", password, token)
        print(f"authorization token: {token}")
        subprocess.run("whoami", shell=True)
        return {
            "table_name": table_name,
            "sql": sql,
            "delete_sql": delete_sql,
            "api_key": api_key,
        }

File: examples/csharp-security-fail/src/Application/Orders/OrderQueryService.cs
using Microsoft.Extensions.Logging;
using System.Diagnostics;

namespace Sample.Application.Orders;

public sealed class OrderQueryService
{
    private readonly ILogger<OrderQueryService> logger;
    private readonly string tableName = "Orders";

    public OrderQueryService(ILogger<OrderQueryService> logger)
    {
        this.logger = logger;
    }

    public void Execute(string userId, string password, string token)
    {
        var sql = "SELECT * FROM Orders WHERE UserId = '" + userId + "'";
        var deleteSql = $"DELETE FROM Orders WHERE UserId = '{userId}'";
        string apiKey = "hard-coded-secret";
        logger.LogInformation("login password={password} token={token}", password, token);
        Console.WriteLine($"authorization token: {token}");
        Process.Start("cmd.exe", "/c whoami");
        _ = new { sql, deleteSql, apiKey, tableName };
    }
}
